MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -486,8 +486,18 @@ public ContractMarket getContractMarket() {
     	return contractMarket;
     }
 
-    public void generateNewContractMarket() {
-    	contractMarket.generateContractOffers(this);
+    // This function clears the campaign's existing contract market
+    // and generates a brand new instance based on the campaign's current faction
+    // use with caution :)
+    public void initializeNewContractMarket() {
```

Comment:
Maybe at the moment it is, but keep in mind that sometimes that's done on purpose for future expansion/planning or to be accessed when the program is used as a library